test/SILOptimizer/array_contentof_opt.swift

```diff
@@ -1,4 +1,4 @@
-// RUN: %target-swift-frontend -O -sil-verify-all -emit-sil -enforce-exclusivity=unchecked  %s | %FileCheck %s
+// RUN: %target-swift-frontend -O -sil-verify-all -emit-sil -enforce-exclusivity=unchecked -Xllvm '-sil-inline-never-functions=$sSa6appendyyxn' %s | %FileCheck %s
```

Comment:
The purpose of this test is to check if the array.append(contentof) optimization works.
With your change it does not work anymore. Adding this "workaround" option to let the test pass does not make any sense.

Reply:
@eeckstein 
As the commit log said:
fix a test case that assumes Array.append is never inlined.

I just added another comment to make it more clear:
// The checks lines are written assuming that the array(Element) calls
// are never inlined. Consequently, the RUN line must prevent inlining
// Array(Element).

What makes you think this PR broke an optimization?

Reply:
As written in the comment in the test file: This is an end-to-end test of the array(contentsOf) -> array(Element) optimization

This optimization replaces an Array.append(contentof:) with a sequence of Array.append(element).
The point of this test is to check if the optimization works. And working means: out of the box without any special command line options.

Reply:
@eeckstein I don't understand why you think the optimization is broken. If array(Element) is inlined after the optimization, then the test's CHECK lines will fail. So it wouldn't be an effective end-to-end test. A proper end-to-end test is going to need to control inlining if it's checking for the presence of calls.

Reply:
Sorry, I misunderstood the change here. Looks good!

Reply:
I cleared up the test case comments
https://github.com/apple/swift/pull/22445/commits/1e73e548caabf8bcf8d598105adadaea67f4d594

Thanks